test/jobs/connector_playbook_execution_reporter_task_test.rb

```diff
@@ -77,20 +79,25 @@ def send_report(report)
 
     ArrangeTestHost.any_instance.stubs(:job_finished?).returns(false, true)
 
-    actual = ForemanTasks.sync_task(ArrangeTestHost, @job_invocation)
+    action = create_and_plan_action(ArrangeTestHost, @job_invocation)
+    action = run_action(action)
 
-    actual_report1 = actual.output[:saved_reports].first.to_s
-    actual_report2 = actual.output[:saved_reports].second.to_s
+    # Process polling cycles - manually trigger Poll events until done
+    run_action(action, Dynflow::Action::Polling::Poll) until action.done?
```

Comment:
The assignment is sort of important. While run_action may modify the action's input and output in place, it doesn't modify action's metadata. It might have no effect on this particular test, but I'd prefer to strive for correctness

```suggestion
    action = run_action(action, Dynflow::Action::Polling::Poll) until action.done?
```

Reply:
Sounds good. I wasn't sure what `run_action` returned, but if it returns the action then all good.